tests/run/errorhandling/Result.scala

```diff
@@ -0,0 +1,70 @@
+package dotty.util
+import boundary.Label
+
+abstract class Result[+T, +E]
+case class Ok[+T](value: T)  extends Result[T, Nothing]
+case class Err[+E](value: E) extends Result[Nothing, E]
```

Comment:
Would it be more natural to call this `Bad` rather than `Err`? As `Err(or)` is not an opposite or an antonym of `Ok`. It also makes it feel like value `E` should extend `Exception` or `Error`.

Read out loud the following:
- "This is an Ok Result"
- "This is an Err Result"
- "This is a Bad Result"

Reply:
- "This is a Fine Result"
- "This is a Fail Result".

Reply:
@ScalaWilliam , type E should stay open, a lot of the times it won't be an exception but rather a case class or a string.
i.e.
`if nameStr.isBlank then Err("name is empty") else Ok(nameStr)`

Reply:
@ScalaWilliam 

[Scalactic](https://www.scalactic.org/user_guide/OrAndEvery) uses `Good` and `Bad`, "good" is the direct antonym of "bad". On the other side, using `Bad` for `Result` might cause some conflicts in code using Scalactic, or at least lead to some confusion. While this should not drive the design of the standard library, I think it's worth considering what are the names used in popular libraries. I did not see `Ok` or `Err` in any other validation library, and their meaning seems already obvious.

Scalactic's `Bad` and cats' `Validated` are not bound to `Exception` or any other error type, I think `Err` should do the same if we want it to replace them.